src/index.spec.js

```diff
@@ -281,6 +296,27 @@ describe('Dropzone', () => {
       expect(dropzone.state().isDragActive).toEqual(true);
       expect(dropzone.state().isDragReject).toEqual(false);
     });
+
+    it('should expose state to children', () => {
+      const dropzone = mount(
+        <Dropzone accept="image/*">
+          {({ isDragActive, isDragReject }) => {
+            if (isDragReject) {
+              return <div id="rejected">Rejected</div>;
+            }
+            if (isDragActive) {
+              return <div id="active">Active</div>;
+            }
+            return <div id="blank">some content</div>;
+          }}
+        </Dropzone>
+      );
+      expect(dropzone.find('#blank')).not.toBeUndefined();
+      dropzone.simulate('dragEnter', { dataTransfer: { files: images } });
+      expect(dropzone.find('#active')).not.toBeUndefined();
+      dropzone.simulate('dragEnter', { dataTransfer: { files } });
+      expect(dropzone.find('#rejected')).not.toBeUndefined();
```

Comment:
I think it would be interesting to also test that `#active` and `#blank` is removed in this case. So for each case we would like to ensure only one of these 3 possible divs is rendered.